### PR TITLE
Change PegoutData to use bitcoin::Script (pending rust-bitcoin v0.26)

### PR DESCRIPTION
**Pull request by shesek**

### CODE REVIEW

**Comment by shesek** [comment]
Ugh, it looks like I somehow missed some failing tests. I was certain I saw them passing earlier... >_< Will look into fixing this.

**Comment by shesek** [comment]
The culprit was a test using `hex_deserialize!()` for the pegout script pubkey. I fixed this in https://github.com/ElementsProject/rust-elements/pull/69/commits/5eba47fa6e49ab12a324a3be5e58645603047b86 using `bitcoin::Script::from_hex()`, but this requires rust-bitcoin v0.26. I tried locally and it appears to upgrade cleanly with no changes. Should I send a PR for this?

File: src/transaction.rs [hunk 1]
```diff
@@ -1274,9 +1274,9 @@ mod tests {
                 genesis_hash: bitcoin::BlockHash::from_hex(
                     "0f9188f13cb7b2c71f2a335e3a4fc328bf5beb436012afca590b1a11466e2206"
                 ).unwrap(),
-                script_pubkey: hex_deserialize!(
-                    "1976a914bedb324be05d1a1254afeb3e7ef40fea0368bc1e88ac"
-                ),
+                script_pubkey: bitcoin::Script::from_hex(
+                    "76a914bedb324be05d1a1254afeb3e7ef40fea0368bc1e88ac"
```
Comment (shesek) [inline src/transaction.rs:1278]: The `0x19` length prefix was needed for `encode::deserialize()`, but isn't when directly constructing a `Script`.

**Review by apoelstra — APPROVED** [review]
ack b8c32797d1766fdd96131e5e77c1eb167a01fdb6

**Pull request merged**